pandas/core/groupby.py

```diff
@@ -247,15 +247,18 @@ def _get_grouper(self, obj):
                                                           sort=self.sort)
         return self.binner, self.grouper, self.obj
 
-    def _set_grouper(self, obj, sort=False):
+    def _set_grouper(self, obj, sort=False, converter=None):
         """
         given an object and the specifications, setup the internal grouper
         for this particular specification
 
         Parameters
         ----------
         obj : the subject object
-
+        sort : bool, default False
+            whether the resulting grouper should be sorted
+        converter : callable, optional
```

Comment:
The root (existing) problem is that sometimes `PeriodIndex` resampling falls back to converting to `DatetimeIndex` - [here](https://github.com/pydata/pandas/blob/master/pandas/tseries/resample.py#L766).  

In the case where we are not using the index, but a selection, there needs to be a way to convert the selected column or level.  I'm using a callback to the selection function to get there - do you see a better way?